### DISCUSSION TITLE
How to I convert a custom sklearn transformer that uses a Tensorflow embedding model?

### DISCUSSION THREAD

**Discussion by shezadkhan137** [onnx/onnx#5695] [Q&A]
Hi, I have a sklearn pipeline that uses a transformer that calls a tensorflow embedding model (USEv4):

```python
class TextEmbeddingTransformer(BaseEstimator, TransformerMixin):

    def __init__(self):
        self.embedding_model = tensorflow_hub.load("https://tfhub.dev/google/universal-sentence-encoder/4")

    def fit(self, X, y=None):
        return self

    def transform(self, X):
        return self.embedding_model(X).numpy()
```

I then use it in a pipeline:

```python
pipeline = Pipeline([
        ('preprocessing', ColumnTransformer([
            ('vectorizer', CountVectorizer(), 'text_one'),
            ('content_embedding', TextEmbeddingTransformer(), 'text_two')
        ], sparse_threshold=0)),
        ('model', LGBMClassifier())
    ])
```

I'm struggling to figure out how to convert the whole pipeline to onnx. I know I can convert the embedding model itself to onnx, but I'm not sure what to do from then.

**Comment by xadupre** [comment]
You may create a custom converter mapped to your TextEmbeddingTransformer (see http://onnx.ai/sklearn-onnx/auto_tutorial/plot_icustom_converter.html) and append the onnx graph coming from tf2onnx. A full example is implemented in one unit test https://github.com/onnx/sklearn-onnx/blob/main/tests/test_other_converter_library_pipelines.py.
